src/ESM.t.sol

```diff
@@ -296,4 +296,34 @@ contract ESMTest is DSTest {
 
         usr.callJoin(esm, 10);
     }
+
+    function test_file_new_min() public {
+        esm = new ESM(address(gem), address(end), address(this), 10);
+        assertEq(esm.min(), 10);
+
+        esm.file("min", 20);
+
+        assertEq(esm.min(), 20);
+    }
+
+    function testFail_file_revoked_gov() public {
+        esm = new ESM(address(gem), address(end), address(0), 10);
+        assertEq(esm.min(), 10);
+
+        esm.file("min", 20);
+    }
+
+    function testFail_file_not_gov() public {
+        esm = new ESM(address(gem), address(end), pauseProxy, 10);
+        assertEq(esm.min(), 10);
+
+        esm.file("min", 20);
+    }
+    
+    function testFail_file_no_min() public {
+        esm = new ESM(address(gem), address(end), address(this), 10);
+        assertEq(esm.min(), 10);
+
+        esm.file("min", 0);
+    }
```

Comment:
Maybe add a testFail for the unrecognized param case, just to cover all the branches

Reply:
You could also add a test `file_new_min_then_fire` where a user tries to `fire()` with just enough MKR to meet the previous `min` and fails

Reply:
:+1:

Reply:
Beautiful.